### DISCUSSION TITLE
How to add MySQL 8 in Laragon Portable?

### DISCUSSION THREAD

**Discussion by OmarHasan16** [leokhoa/laragon#928] [General]
Adding MySQL 8 to Laragon Portable

### Step 1: Download MySQL 8
1. Visit the official [[MySQL Community Downloads page](https://dev.mysql.com/downloads/mysql/)](https://dev.mysql.com/downloads/mysql/).
2. Select the platform (Windows) and download the **ZIP Archive** version of MySQL 8.
3. Extract the downloaded archive to a directory (e.g., `C:\laragon\bin\mysql\mysql-8.x.x`).

### Step 2: Integrate MySQL 8 into Laragon
1. Navigate to your Laragon installation directory.
2. Go to the `bin\mysql` folder (e.g., `C:\laragon\bin\mysql`).
3. Place the extracted MySQL folder (e.g., `mysql-8.x.x`) inside the `mysql` directory.

### Step 3: Update MySQL Configuration in Laragon
1. Open Laragon.
2. Go to **Menu** → **Preferences** → **Services & Ports**.
3. In the **Database** section, set the path to the new MySQL 8 version if necessary.
4. If you want to configure MySQL further:
   - Open the `my.ini` configuration file located in the `mysql-8.x.x` folder.
   - Update paths or settings as needed.

### Step 4: Switch to MySQL 8 in Laragon
1. Restart Laragon to detect the new MySQL version.
2. In Laragon's main window, click on the **Database** section (next to Start All).
3. Select **MySQL 8.x.x** from the drop-down menu.

### Step 5: Test the Installation
1. Start Laragon services.
2. Use a database client (e.g., HeidiSQL or Laragon's built-in terminal) to connect to MySQL 8.
3. Execute `SELECT VERSION();` to confirm MySQL 8 is running.

But not working

**Comment by Sophist-UK** [comment]
At this point, with a new version being released apparently tomorrow, we should I think hope that the default MySQL version is already 8 and that these instructions will be moot.

I am hopeful that the new version will do a lot of what I have been waiting for, including having MySQL 8 as a default version, and that I can continue to use Laragon and to be an active advocate for others to use it.

However, if it turns out that the next release doesn't have a more modern version of MySQL, especially after such a long long long wait for a new version, that is going to severely dent my confidence in Laragon as a solution, and I will probably switch to FlyEnv (which I am already actively evaluating).
